torch/csrc/distributed/c10d/control_plane/Handlers.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <functional>
+#include <string>
+
+namespace c10d {
+namespace control_plane {
+
+class Request {
+ public:
+  virtual ~Request() = default;
+
+  virtual const std::string& body() = 0;
+};
+class Response {
+ public:
+  virtual ~Response() = default;
+
+  virtual void setContent(
+      std::string&& content,
+      const std::string& content_type) = 0;
+  virtual void setStatus(int status) = 0;
+};
+
+using HandlerFunc = std::function<void(const Request&, Response&)>;
+
+void registerHandler(const std::string& name, HandlerFunc f);
+HandlerFunc getHandler(const std::string& name);
+
+// Registers a handler statically.
+class RegisterHandler {
+ public:
+  RegisterHandler(const std::string& name, HandlerFunc f) {
+    registerHandler(name, f);
```

Comment:
Is it a standard pattern to update shared state in a constructor?

Reply:
Yes, this is a static object guard -- we use the same thing for torch custom op registration. The shared state is in a meyers singleton in getHandlersRegistry so it neatly avoids any static initialization ordering issues.

Ex: https://github.com/pytorch/pytorch/blob/ed327876f574c6a014321be419a3ae9808a765a2/torch/csrc/jit/runtime/custom_operator.h#L21